### PR TITLE
Fix #235: Pragma displacing real statements

### LINKED ISSUE #235: Incorrect AST Structure when #pragma follows For loop.
Consider the following code snippet: 
```c
for(int i = 0; i < 3; i++)
    #pragma omp critical
        sum += 1;
```
When compiled without Open MP, the #pragma should be ignored completely, so that the statement ``sum += 1`` should be a descendent of the for loop. However, in the current implementation of pycparser, it is parsed a _sibling_ of the for loop, instead of as a descendant. 
```
For: 
  DeclList: 
    Decl: i, [], [], []
      TypeDecl: i, []
        IdentifierType: ['int']
      Constant: int, 0
  BinaryOp: <
    ID: i
    Constant: int, 3
  UnaryOp: p++
    ID: i
  Pragma: omp critical
Assignment: +=
  ID: sum
  Constant: int, 1
```

The same problem applies to other loops, Labels, and If statements, as in the following:
```c
for(int i = 0; i < 3; i++)
    myLabel:
    #pragma omp critical
        sum += 1;
```
```c
while(sum < 100)
    #pragma omp critical
        sum += 1;
```
```c
if (sum < 100)
    #pragma omp critical
    sum += 1;
```
The following will not even parse, but it should:
```c        
do 
    #pragma omp critical
    sum += 1;
while(sum < 100)
```

### PR DESCRIPTION
**Pull request by dbluhm**
I apologize if this is terribly naive but I believe this should take care of #235 by wrapping pragma's and the directly following statement in a Compound. As discussed in #235 this is kind of weird but appears to be the best way to ensure (mostly) correct AST structure in these cases.

### CODE REVIEW

**Comment by dbluhm** [comment]
I failed to run tests before submitting :stuck_out_tongue_closed_eyes:. I'm looking into the errors reported now.

**Comment by dbluhm** [comment]
One issue was that the there was no default case for pragma's without statements. Now, however, tests fail due to the structure being slightly different. @eliben, what do you think of these changes?

**Comment by dbluhm** [comment]
Narrowed the affected pragma's with b1ff55d

File: pycparser/c_parser.py [hunk 1]
```diff
@@ -616,6 +616,17 @@ def p_statement(self, p):
         """
         p[0] = p[1]
 
+    def p_pragmacomp_or_statement(self, p):
+        """ pragmacomp_or_statement	: pppragma_directive statement
+                                        | statement
```
Comment (eliben) [inline pycparser/c_parser.py]: Align the | with the : on the line above

File: pycparser/c_parser.py [hunk 2]
```diff
@@ -616,6 +616,17 @@ def p_statement(self, p):
         """
         p[0] = p[1]
 
+    def p_pragmacomp_or_statement(self, p):
+        """ pragmacomp_or_statement	: pppragma_directive statement
```
Comment (eliben) [inline pycparser/c_parser.py]: Can the pragma come after the statement?

File: pycparser/c_parser.py [hunk 3]
```diff
@@ -616,6 +616,17 @@ def p_statement(self, p):
         """
         p[0] = p[1]
 
+    def p_pragmacomp_or_statement(self, p):
```
Comment (eliben) [inline pycparser/c_parser.py]: This needs a comment explaining what's going on, and also tests

File: pycparser/c_parser.py (hunk 2 above)
Comment (dbluhm) [reply pycparser/c_parser.py]: I can't think of a situation where a statement followed by a pragma would be a case for grouping with a compound. The pragma is the decorator for the following statement but the statement preceding a pragma wouldn't mean anything to the pragma.

**Pull request merged**